[rocsparse] Add Disclaimers regarding hipSPARSE (#943)
ROCm/rocm-libraries: projects/rocsparse/docs/index.rst

rocSPARSE documentation
********************************************************************

.. important::
   **hipSPARSE is the recommended library for development**
   

>    For new development projects, we strongly recommend using **hipSPARSE** instead of rocSPARSE. 

Reviewer: ```suggestion
   For new development projects, **hipSPARSE** is recommended instead of rocSPARSE. 
```

   hipSPARSE provides a comprehensive, portable interface that supports multiple backends 
   (including rocSPARSE and cuSPARSE) and is the preferred library for sparse matrix operations.
   

>    For documentation and examples, please refer to the `hipSPARSE documentation <https://rocm.docs.amd.com/projects/hipSPARSE/en/latest/>`_.

Reviewer: ```suggestion
   For documentation and examples, see the `hipSPARSE documentation <https://rocm.docs.amd.com/projects/hipSPARSE/en/latest/>`_.
```

>    This rocSPARSE documentation is maintained for reference purposes and to document any 

Reviewer: ```suggestion
   The rocSPARSE documentation is maintained for reference purposes and to document any 
```

>    specialized functionality that may not be available in hipSPARSE.

Reviewer: ```suggestion
   specialized functionality that might not be available in hipSPARSE.
```
